storage/remote/queue_manager.go

```diff
@@ -446,19 +446,25 @@ func (s *shards) runShard(i int) {
 			queueLength.WithLabelValues(s.qm.queueName).Dec()
 			pendingSamples = append(pendingSamples, sample)
 
+			sentSamples := len(pendingSamples) >= s.qm.cfg.MaxSamplesPerSend
+
 			for len(pendingSamples) >= s.qm.cfg.MaxSamplesPerSend {
 				s.sendSamples(pendingSamples[:s.qm.cfg.MaxSamplesPerSend])
 				pendingSamples = pendingSamples[s.qm.cfg.MaxSamplesPerSend:]
```

Comment:
Just  for readability I'd make sentSamples = true here, and avoid the complex condition above the loop. Otherwise, this makes sense, thanks for pointing this out.

Reply:
Done, thanks for the feedback!